src/ray/raylet/scheduling_policy.cc

```diff
@@ -116,36 +118,54 @@ std::unordered_map<TaskID, ClientID> SchedulingPolicy::Schedule(
   return decision;
 }
 
-std::vector<TaskID> SchedulingPolicy::SpillOver(
-    SchedulingResources &remote_scheduling_resources) const {
+std::unordered_map<TaskID, ClientID> SchedulingPolicy::SpillOver(
+    std::unordered_map<ClientID, SchedulingResources> &cluster_resources,
+    std::vector<ClientID> &remote_client_ids) {
+  // Shuffle the nodes to prevent infinite forwarding loops.
+  const unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
+  auto rng = std::default_random_engine(seed);
+  std::shuffle(remote_client_ids.begin(), remote_client_ids.end(), rng);
```

Comment:
Can the shuffle be done in the monitor, or does it need to be performed independently at every node?

Reply:
It can be done in the monitor (which I tested and found to work) but my reasoning was that it doesn't make sense to shuffle there since the monitor isn't responsible for fixing a scheduling policy issue. But yeah if I move it there it does work.

Reply:
I think we should probably put the shuffle in the monitor for now and avoid changes to the scheduling policy in this PR.